zipline/gens/tradesimulation.py

```diff
@@ -272,14 +272,20 @@ def past_auto_close_date(asset):
         for asset in assets_to_clear:
             perf_tracker.process_close_position(asset, dt, data_portal)
 
-        # Remove open orders for any sids that have reached their
-        # auto_close_date.
+        # Remove open orders for any sids that have reached their auto close
+        # date. These orders get processed immediately because otherwise they
+        # would not be processed until the first bar of the next day.
         blotter = algo.blotter
-        assets_to_cancel = \
-            set([asset for asset in blotter.open_orders
-                 if past_auto_close_date(asset)])
+        assets_to_cancel = [
+            asset for asset in blotter.open_orders
+            if past_auto_close_date(asset)
+        ]
         for asset in assets_to_cancel:
             blotter.cancel_all_orders_for_asset(asset)
+        for order in copy(blotter.new_orders):
```

Comment:
Inside this loop I remove elements from the `blotter.new_orders` list, so I had to make a copy such that I wasn't modifying the same list I was iterating over.